### PR TITLE
HPCC-29175 Document HPCC_DEPLOYMENT environment variable

### PR DESCRIPTION
**Pull request by JamesDeFabia**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:

http://10.224.20.18/view/Docs-gp/job/DocBuild-JD01/19/artifact/build-docs/Release/docs/EN_US/ECLLanguageReference_EN_US-9.0.3-closedown0.pdf

http://10.224.20.18/view/Docs-gp/job/DocBuild-JD01/19/artifact/build-docs/Release/docs/EN_US/HPCCSystemAdministratorsGuide_EN_US-9.0.3-closedown0.pdf

### CODE REVIEW

**Review by RichardTaylorHPCC — APPROVED** [review]
Looks good

File: docs/EN_US/HPCCSystemAdmin/HPCCSystemAdministratorsGuide.xml [hunk 1]
```diff
@@ -1487,8 +1480,7 @@ lock=/var/lock/HPCCSystems</programlisting>
                   </row>
                 </tbody>
               </tgroup>
-            </informaltable>
-          </para>
+            </informaltable></para>
```
Comment (jakesmith) [inline docs/EN_US/HPCCSystemAdmin/HPCCSystemAdministratorsGuide.xml:1483]: the indentation is bit odd here, doesn't line up with the opening <informaltable> or the <para> on line 1466

File: docs/EN_US/HPCCSystemAdmin/HPCCSystemAdministratorsGuide.xml [hunk 2]
```diff
@@ -41,7 +41,7 @@
       similarity to actual persons, living or dead, is purely
       coincidental.</para>
 
-      <para />
+      <para></para>
```
Comment (jakesmith) [inline docs/EN_US/HPCCSystemAdmin/HPCCSystemAdministratorsGuide.xml:44]: question: is this line needed/wanted?

**Review by jakesmith — APPROVED** [review]
Content looks fine, trivial formatting issues maybe.

**Review by g-pan — APPROVED** [review]
OK to merge

File: docs/EN_US/ECLLanguageReference/ECLR_mods/Basics-EnvironmentVariables.xml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+    <para><programlisting>helm install myenv1 hpcc/hpcc</programlisting></para>
+
+    <para>then the HPCC_DEPLOYMENT variable will contain the value: myenv1.
+    </para>
+
+    <para>For a bare-metal environment, this value can be set in
+    environment.conf, by adding:</para>
+
+    <para><programlisting>deploymentName: myenv1</programlisting></para>
```
Comment (ghalliday) [inline docs/EN_US/ECLLanguageReference/ECLR_mods/Basics-EnvironmentVariables.xml:25]: This looks wrong.  Shouldn't it be

`deploymentName=myenv1
`
?

**Review by ghalliday — COMMENTED** [review]
@JamesDeFabia one comment.
Also, how can we stop the unrelated changes that seem to often occur when a file is edited?  It makes it harder to spot the content that has actually changed.

File: docs/EN_US/ECLLanguageReference/ECLR_mods/Basics-EnvironmentVariables.xml (hunk 3 above)
Comment (JamesDeFabia) [reply docs/EN_US/ECLLanguageReference/ECLR_mods/Basics-EnvironmentVariables.xml]: @ghalliday I made the edit and pushed. Please look again

**Review by ghalliday — APPROVED** [review]

**Pull request merged**